release/mkrelease.js

```diff
@@ -0,0 +1,306 @@
+        }
+        callback();
+    });
+}
+
+function writeAgentVersionFile(newRelease)
+{
+    console.log("Writing agent version file")
+    if (!opt.options.dryrun)
+    {
+        fs.writeFileSync(__dirname + '/../src/agentversion', newRelease  + "\n");
+    }
+    return newRelease;
+}
+
+function fetchPRsSinceLastReleaseAndEditReleaseNotes(newRelease, callback)
+{
+    var derivedFrom = opt.options.derivedFrom;
+    console.log("Derived from %o", derivedFrom);
+    if (derivedFrom !== 'latest')
+    {
+        if (!derivedFrom.startsWith('v'))
+        {
+            derivedFrom = 'v' + derivedFrom;
+        }
+        derivedFrom = 'tags/' + derivedFrom;
+    }
+    gitHubRequest(gitHubAPIURLRoot + "/releases/" + derivedFrom, { json: true }, function (err, resp, body) {
+        if (err) throw err;
+        if (body.published_at === undefined)
+        {
+            console.log('Error: Cannot find release ' + opt.options.derivedFrom + '. Aborting.');
+            process.exit(-1);
+        }
+        var lastReleaseDate = body.published_at;
+        console.log("Fetching PRs merged since " + lastReleaseDate);
+        gitHubRequest("https://api.github.com/search/issues?q=type:pr+is:merged+repo:microsoft/azure-pipelines-agent+merged:>=" + lastReleaseDate + "&sort=closed_at&order=asc", { json: true }, function (err, resp, body) {
+            editReleaseNotesFile(body, callback);
+        });
+    });
+}
+
+function editReleaseNotesFile(body, callback)
+{
+    var releaseNotesFile = __dirname + '/../releaseNote.md';
+    var existingReleaseNotes = fs.readFileSync(releaseNotesFile);
+    var newPRs = [];
+    body.items.forEach(function (item) {
+        newPRs.push(' - ' + item.title + ' (#' + item.number + ')');
+    });
+    var newReleaseNotes = newPRs.join("\n") + "\n\n" + existingReleaseNotes;
+    var editorCmd = process.env.EDITOR + ' ' + releaseNotesFile;
+    console.log(editorCmd);
+    if (opt.options.dryrun)
+    {
+        console.log("Found the following PRs = %o", newPRs);
+    }
+    else
+    {
+        fs.writeFileSync(releaseNotesFile, newReleaseNotes);
+        try
+        {
+            cp.execSync(process.env.EDITOR + ' ' + releaseNotesFile, {
+                stdio: [process.stdin, process.stdout, process.stderr]
+            });
+        }
+        catch (err)
+        {
+            console.log(err.message);
+            process.exit(-1);
+        }
+    }
+    callback();
+}
+
+function versionifySync(template, destination, version)
+{
+    try
+    {
+        var data = fs.readFileSync(template, 'utf8');
+        data = data.replace(/<AGENT_VERSION>/g, version);
+        console.log("Generating " + destination);
+        fs.writeFileSync(destination, data);
+    }
+    catch(e)
+    {
+        console.log('Error:', e.stack);
+    }
+}
+
+function createIntegrationFiles(newRelease, callback)
+{
+    mkpath(integrationDir, function (err) {
```

Comment:
Rather than going async and adding this `mkpath` module, can we just use node's built in [mkdirSync](https://nodejs.org/api/fs.html#fs_fs_mkdirsync_path_options) with the recursive option? You need node 10 for the recursive option, but that's also current LTS